Sasquatch/SasquatchObjC/AppDelegate.m

```diff
@@ -383,7 +389,33 @@ - (BOOL)distribute:(MSDistribute *)distribute
   return NO;
 }
 
-#pragma mark - MSPushDelegate
+#pragma mark - Push callbacks
+
+// iOS 10 and later, called when a notification is delivered to an app that is
+// in the foreground.
+// When this callback is called, this disables the other callback that MSPush
+// handles.
+- (void)userNotificationCenter:(UNUserNotificationCenter *)center
+       willPresentNotification:(UNNotification *)notification
+         withCompletionHandler:
+             (void (^)(UNNotificationPresentationOptions options))
+                 completionHandler API_AVAILABLE(ios(10.0)) {
+  self.didReceiveNotificationInForeground = YES;
+  [MSPush didReceiveRemoteNotification:notification.request.content.userInfo];
+  completionHandler(UNNotificationPresentationOptionNone);
+}
+
+// iOS 10 and later, asks the delegate to process the user's response to a
+// delivered notification.
+- (void)userNotificationCenter:(UNUserNotificationCenter *)center
+    didReceiveNotificationResponse:(UNNotificationResponse *)response
+             withCompletionHandler:(void (^)(void))completionHandler
+    API_AVAILABLE(ios(10.0)) {
+  self.notificationResponse = [response actionIdentifier];
```

Comment:
Since [UNNotificationDismissActionIdentifier](https://developer.apple.com/documentation/usernotifications/unnotificationdismissactionidentifier)
> is delivered only if the notification’s category object was configured with the customDismissAction option

and there are only 2 system actions + we are not supporting categories with custom actions, it looks redundant.
Lets just set flag to `NO` here:
```objc
self.didReceiveNotificationInForeground = NO;
```

Reply:
Do you mean adding a new flag `self.userDidTapNotification` instead of the `notificationResponse` ?

Reply:
Just setting `didReceiveNotificationInForeground` to `NO` would be enough

Reply:
If background the notification is always tapped.
If foreground the notification is never tapped.
We could indeed simplify this + the docs.
Let's keep the boolean for both app and docs.